File: app/Modules/Knowledge/KB/compile/src/mpi_hello.c
/*  C  */

#include <stdio.h>
#include <mpi.h>

int main (int argc, char *argv[]) {

    /* MPI Parameters */
    int rank, size, len;
    char name[MPI_MAX_PROCESSOR_NAME];

    /* All ranks initiate the message-passing environment. */
    /* Each rank obtains information about itself and its environment. */
    MPI_Init(&argc, &argv);                 /* start MPI           */
    MPI_Comm_size(MPI_COMM_WORLD, &size);   /* get number of ranks */
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);   /* get rank            */
    MPI_Get_processor_name(name, &len);     /* get run-host name   */

    printf("Runhost:%s   Rank:%d of %d ranks   hello, world\n", name,rank,size);

    MPI_Finalize();                         /* terminate MPI       */
    return 0;
}

File: app/Modules/Knowledge/KB/compile/src/mpi_hello.f
C  Fortran 77

      PROGRAM HELLO

C     MPI Parameters
      INCLUDE 'mpif.h'
      INTEGER RANK, SIZE, IERROR, LEN
      CHARACTER*30 NAME

C     All ranks initiate the message-passing environment.
C     Each rank obtains information about itself and its environment.
C     Start MPI.
      CALL MPI_INIT(IERROR)
C     Get number of ranks.
      CALL MPI_COMM_SIZE(MPI_COMM_WORLD, SIZE, IERROR)
C     Get rank.
      CALL MPI_COMM_RANK(MPI_COMM_WORLD, RANK, IERROR)
C     Get run-host name.
      CALL MPI_GET_PROCESSOR_NAME(NAME,LEN,IERROR)

      PRINT *, 'Runhost:', name, '   Rank:', RANK, ' of ', SIZE,
     &         'ranks', '   hello, world'

C     Terminate MPI.
      CALL MPI_FINALIZE(IERROR)
      END